vision_people_logging/scripts/save_ubd.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+import rospy
+from mongodb_store.message_store import MessageStoreProxy
+from topological_logging_manager.msg import LoggingManager
+from upper_body_detector.msg import UpperBodyDetector
+from vision_people_logging.msg import LoggingUBD
+import geometry_msgs.msg
+import sensor_msgs.msg
+import message_filters
+from cv_bridge import CvBridge
+import tf
+
+
+class SaveLocations():
+    def __init__(self, collname="upper_bodies"):
+        self.collname = collname
+        rospy.loginfo("Intialising UBD logging")
+        self.robot_pose = geometry_msgs.msg.Pose()
+        self.tfl = tf.TransformListener()
+        self.msg_store = MessageStoreProxy(collection=collname)
+        self.counter = 0
+
+        # Always get the robot's position.
+        rospy.Subscriber("/robot_pose", geometry_msgs.msg.Pose, self.posecb, None, 1)
+
+        subs = [
+            message_filters.Subscriber(rospy.get_param("~ubd", "/upper_body_detector/detections"), UpperBodyDetector),
+            message_filters.Subscriber(rospy.get_param("~ubd_cent", "/upper_body_detector/bounding_box_centres"), geometry_msgs.msg.PoseArray),
+            message_filters.Subscriber(rospy.get_param("~ubd_rgb", "/head_xtion/rgb/image_rect_color"), sensor_msgs.msg.Image),
+            message_filters.Subscriber(rospy.get_param("~ubd_d", "/head_xtion/depth/image_rect_meters"), sensor_msgs.msg.Image),
+        ]
+
+        # Potentially also make use of the manager to tell us whether
+        # we are allowed to record or not.
+        mgr_topic = rospy.get_param("~manager", None)
+        if mgr_topic is not None:
+            subs += [message_filters.Subscriber(mgr_topic, LoggingManager)]
+
+        ts = message_filters.ApproximateTimeSynchronizer(subs, queue_size=5, slop=0.15)
+        ts.registerCallback(self.cb)
+
+
+    def posecb(self, pose):
+        self.robot_pose = pose
+
+
+    def cb(self, ubd, ubd_cent, rgb, d, *mgr):
+        # Check for perission, if necessary:
+        if len(mgr) and not mgr[0].log:
+            return
```

Comment:
Thinking about it, this bypasses one of the safety features of the logging manager. If it should be running, but died this will not return. So this is true if it running and publishes `true` and if it is not running but should be.

The synchronised callback should not be triggered in the latter case. Maybe just checking if the topic has been set instead of `len(mgr)` would work.